compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -126,6 +138,11 @@ impl<'a> FunctionContext<'a> {
         } else {
             self.builder.new_function(func.name.clone(), id, func.inline_type);
         }
+
+        for (_, value) in self.shared_context.globals_context.dfg.values_iter() {
+            self.builder.current_function.dfg.make_global(value.get_type().into_owned());
+        }
```

Comment:
This snippet has popped up here twice, and in normalize_value_ids IIRC. It's non-trivial enough that I think we should move it into a helper function.

Reply:
done